capsules/src/dals/interfaces.rs

```diff
@@ -0,0 +1,107 @@
+//! All interfaces between capsules for `DALS`
+
+/// Errors which prevent an app from being loaded into memory. 
+pub enum LoadingError {
+    //TODO
+}
+
+/// Errors thrown by the `Decompressor` (errors which prevent an app from being decompressed)
+pub enum DecompressionError {
+    //TODO
+}
+
+/// Errors thrown by the `Verifier` if the loaded app did not pass the verification.
+pub enum VerificationError {
+    //TODO
+}
+
+/// Errors thrown by the `Validator` if the loaded app did not pass the validation.
+pub enum ValidationError {
+    //TODO
+}
+
+/// Implemented by any module (client) which seeks to load data buffer-by-buffer from a hardware resource
+/// such as Wifi, Bluetooth, Zigbee, USB, etc.
+pub trait AppLoaderClient<'a>{
+    /// Called by `AppLoader` implementation when data buffer is ready to be returned to `AppLoaderClient`
+    ///
+    /// - `data_buffer`: Buffer which was passed to `AppLoader` via the `next_buffer()` call 
+    fn return_buffer(&self, data_buffer: &'static mut [u8]);
+}
+
+/// Implemented by main `AppLoader` module
+pub trait AppLoader<'a>{
```

Comment:
Does it make sense to express the dependence on the client traits here?
```suggestion
pub trait AppLoader<'a>: DecompressorClient<'a> + VerifierClient<'a> + ValidatorClient<'a> {
```

Reply:
Yes. It does.  That is a good idea to make sure they are all implemented by the same module.